### PR TITLE
CNV-46603: UDN list & form implementation

### PR DESCRIPTION
**Pull request by jpinsonneau**
Add `UserDefinedNetwork`:
- [X] List implementation
- [X] YAML example
- [x] Edit form
- [x] Detail page

image
image
image
image

A followup PR will be created for `ClusterUserDefinedNetwork` implementation

### CODE REVIEW

File: src/views/udns/actions/hooks/useUDNActions.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    },
+    {
+      accessReview: asAccessReview(UserDefinedNetworkModel, obj, 'update'),
+      cta: launchAnnotationsModal,
+      id: 'edit-annotations-udn',
+      label: t('Edit annotations'),
+    },
+    {
+      accessReview: asAccessReview(UserDefinedNetworkModel, obj, 'update'),
+      cta: () => navigate(`/k8s/ns/${objNamespace}/${UserDefinedNetworkModelRef}/${objName}/yaml`),
```
Comment (upalatucci) [inline src/views/udns/actions/hooks/useUDNActions.tsx:42]: You can use `resourcePathFromModel` for that

File: src/views/udns/details/tabs/details/UDNLayerDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+  const getList = (children: React.JSX.Element[]) => {
+    if (!children.length) {
+      return getEmptyText();
+    }
+    return <List>{children}</List>;
+  };
+
+  const getContent = () => {
+    switch (getTopology(udn)) {
```
Comment (upalatucci) [inline src/views/udns/details/tabs/details/UDNLayerDetails.tsx:84]: consider creating different components to handle this switch

File: src/utils/resources/udns/utils.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  apiVersion: 'v1',
+  crd: true,
+  id: 'userdefinednetwork',
+  kind: 'UserDefinedNetwork',
+  label: 'User Defined Network',
+  labelPlural: 'User Defined Networks',
+  legacyPluralURL: true,
+  namespaced: true,
+  plural: 'userdefinednetworks',
+};
```
Comment (upalatucci) [inline src/utils/resources/udns/utils.ts:16]: In my pr i created this exact model into `src/utils/models/index.ts`
For me it's okay to have it here but let's just be aware that when both prs will be merged we don't want to different UDN model definition

Comment (jpinsonneau) [reply src/utils/resources/udns/utils.ts:16]: yeah I saw that after creating this PR :smile_cat: 

I will point your definition as soon as your PR is merged

File: src/views/udns/actions/hooks/useUDNActions.tsx (hunk 1 above)
Comment (jpinsonneau) [reply src/views/udns/actions/hooks/useUDNActions.tsx]: used `getResourceURL` instead as I point to YAML page

File: src/views/udns/details/tabs/details/UDNLayerDetails.tsx (hunk 2 above)
Comment (jpinsonneau) [reply src/views/udns/details/tabs/details/UDNLayerDetails.tsx:88]: done

**Pull request merged**